DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs

```diff
@@ -499,8 +499,12 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
             MethodInfo getMethod;
             int nonediableOrdering;
             DevicePropertyMetadata propertyMetadata;
+            PropertyType propertyType;
 
-            Debug.Assert(deviceProperties != null, "deviceProperties is a null reference.");
+            if (deviceProperties == null)
+            {
+                throw new ArgumentNullException("deviceProperties is a null reference.");
```

Comment:
this exception should be thrown when 

> Exception thrown when required device properties are not found

this is not the case here I think, its just the calling method passed a null argument